kolibri/utils/tests/test_server.py

```diff
@@ -53,3 +62,121 @@ def test_windows(self):
     def test_whl(self):
         install_type = server.installation_type()
         assert install_type == "whl"
+
+
+@pytest.fixture
+def dbconnection():
+    fd, filepath = tempfile.mkstemp()
+    connection = create_engine(
+        "sqlite:///{path}".format(path=filepath),
+        connect_args={"check_same_thread": False},
+        poolclass=NullPool,
+    )
+    yield connection
+    os.close(fd)
+    os.remove(filepath)
+
+
+@pytest.fixture
+def storage(dbconnection):
+    s = Storage(dbconnection)
+    yield s
+    s.clear()
+
+
+QUEUE = "pytest"
+
+
+class MockServices(object):
```

Comment:
I think the main weakness with the tests is the use of the MockServices plugin here - we should use the real services plugin, as otherwise we are not guaranteeing anything about how the services plugin will behave in production.

Reply:
Yes I agree with you, let me see how can this be done.